code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -128,13 +131,40 @@ def generate_on_your_data_llm_chain(
             "chat_history": chat_history,
         }
 
+    def generate_source_documents(self, search_results):
+        sources = []
+        for result in search_results:
+            source_url = self._extract_source_url(result.get("source", ""))
+
+            metadata_dict = {
+                "id": result.get("id", ""),
+                "title": result.get("title", ""),
+                "source": source_url,
+                "chunk_id": result.get("chunk_id", ""),
+            }
+            sources.append(
+                Document(
+                    page_content=result["content"],
+                    metadata=metadata_dict,
+                )
+            )
+        return sources
+
+    def _extract_source_url(self, original_source):
+        matches = list(re.finditer(r"https?://", original_source))
+        if len(matches) > 1:
+            second_http_start = matches[1].start()
+            source_url = original_source[second_http_start:]
+        else:
+            source_url = original_source + "_SAS_TOKEN_PLACEHOLDER_"
+        return source_url
+
     def answer_question(self, question: str, chat_history: list[dict], **kwargs: dict):
-        # Retrieve documents as sources
-        sources = self.vector_store.similarity_search(
-            query=question,
-            k=self.env_helper.AZURE_SEARCH_TOP_K,
-            filters=self.env_helper.AZURE_SEARCH_FILTER,
-        )
+        if self.env_helper.AZURE_SEARCH_USE_INTEGRATED_VECTORIZATION:
```

Comment:
Now you have the code/backend/batch/utilities/search/Search.py file, I wonder if a new method `retrieve_documents_as_sources` could live there and have this block of code too